packages/flutter/lib/src/widgets/router.dart

```diff
@@ -619,7 +619,7 @@ class _RouterState<T> extends State<Router<T>> {
       });
   }
 
-  static final Future<dynamic> _never = Completer<dynamic>().future; // won't ever complete
+  static final Future<void> _never = Completer<void>().future; // won't ever complete
```

Comment:
I think this one actually has to be dynamic because we cast it to all kinds of different types below.